lineax/_solver/Woodbury.py

```diff
@@ -0,0 +1,138 @@
+from typing import Any
+from typing_extensions import TypeAlias
+
+import jax
+import jax.numpy as jnp
+from jaxtyping import Array, PyTree
+
+from .._operator import (
+    AbstractLinearOperator,
+    is_Woodbury,
+    MatrixLinearOperator,
+    woodbury,
+)
+from .._solution import RESULTS
+from .._solve import AbstractLinearSolver, AutoLinearSolver
+from .misc import (
+    pack_structures,
+    PackedStructures,
+    ravel_vector,
+    transpose_packed_structures,
+    unravel_solution,
+)
+
+
+_Woodbury_State: TypeAlias = tuple[
```

Comment:
Also on naming conventions: this would normally be `_WoodburyState`, to avoid mixing `CamelCase` and `snake_case`.